crates/ruff_db/src/files.rs

```diff
@@ -166,6 +138,28 @@ impl Files {
         Some(file)
     }
 
+    /// Adds a file for the given virtual system path.
+    ///
+    /// It always adds a new file, replacing any existing file for the same path, unless the system
+    /// failed to provide a metadata for the path.
+    #[tracing::instrument(level = "trace", skip(self, db), ret)]
+    pub fn add_virtual_file(&self, db: &dyn Db, path: &SystemVirtualPath) {
+        let Ok(metadata) = db.system().virtual_path_metadata(path) else {
+            return;
+        };
+        self.inner.system_virtual_by_path.insert(
+            path.to_path_buf(),
+            File::new(
+                db,
+                FilePath::SystemVirtual(path.to_path_buf()),
+                metadata.permissions(),
+                metadata.revision(),
+                FileStatus::Exists,
+                Count::default(),
+            ),
+        );
```

Comment:
I think it would be best to not override any existing file. Overriding an existing file would mean that we suddenly have multiple files mapping to the same path. This could be come awkward to debug.